code/modules/mining/mine_turfs.dm

```diff
@@ -1078,3 +1130,139 @@ turf/unsimulated/mineral/ChangeTurf(var/turf/N, var/tell_universe=1, var/force_l
+	if(src.x >= (world.maxx - TRANSITIONEDGE) || src.x <= TRANSITIONEDGE)
+		return BUILD_FAILURE
+	else if (src.y >= (world.maxy - TRANSITIONEDGE || src.y <= TRANSITIONEDGE ))
+		return BUILD_FAILURE
+	else if(!(locate(/obj/structure/lattice) in contents))
+		return BUILD_SUCCESS
+	return BUILD_FAILURE
+
+/turf/simulated/floor/asteroid/canBuildPlating()
+	if(src.x >= (world.maxx - TRANSITIONEDGE) || src.x <= TRANSITIONEDGE)
+		return BUILD_FAILURE
+	else if (src.y >= (world.maxy - TRANSITIONEDGE || src.y <= TRANSITIONEDGE ))
+		return BUILD_FAILURE
+	else if(!dug)
+		return BUILD_IGNORE
+	if(locate(/obj/structure/lattice) in contents)
+		return BUILD_SUCCESS
+	return BUILD_FAILURE
+
+/turf/simulated/floor/asteroid/New()
+	var/proper_name = name
+	..()
+
+	name = proper_name
+
+	if(prob(20))
+		icon_state = "asteroid[rand(0,12)]"
+	if(ticker)
+		initialize()
+
+/turf/simulated/floor/asteroid/initialize()
+	updateMineralOverlays()
+
+/turf/simulated/floor/asteroid/ex_act(severity)
+	switch(severity)
+		if(3.0)
+			return
+		if(2.0)
+			if (prob(70))
+				gets_dug()
+		if(1.0)
+			gets_dug()
+	return
+
+/turf/simulated/floor/asteroid/proc/update_ore_overlays(var/new_ore_overlay)
+	return //Damn ChangeTurf
+
+/turf/simulated/floor/asteroid/attackby(obj/item/weapon/W as obj, mob/user as mob)
+
+	if(!W || !user)
+		return 0
+
+	if (istype(W, /obj/item/weapon/pickaxe))
+		var/obj/item/weapon/pickaxe/used_digging = W //cast for dig speed and flags
+		if (get_turf(user) != user.loc) //if we aren't somehow on the turf we're in
+			return
+
+		if(!(used_digging.diggables & DIG_SOIL)) //if the pickaxe can't dig soil, we don't
+			to_chat(user, "<span class='rose'>You can't dig soft soil with \the [W].</span>")
+			return
+
+		if (dug)
+			to_chat(user, "<span class='rose'>This area has already been dug.</span>")
+			return
+
+		to_chat(user, "<span class='rose'>You start digging.<span>")
+		playsound(get_turf(src), 'sound/effects/rustle1.ogg', 50, 1) //russle sounds sounded better
+
+		if(do_after(user, src, used_digging.digspeed) && user) //the better the drill, the faster the digging
+			playsound(src, 'sound/items/shovel.ogg', 50, 1)
+			to_chat(user, "<span class='notice'>You dug a hole.</span>")
+			gets_dug()
+
+	else
+		..(W,user)
+	return
+
+/turf/simulated/floor/asteroid/proc/gets_dug()
+	if(dug)
+		return
+	new/obj/item/weapon/ore/glass(src)
+	new/obj/item/weapon/ore/glass(src)
+	new/obj/item/weapon/ore/glass(src)
+	new/obj/item/weapon/ore/glass(src)
+	new/obj/item/weapon/ore/glass(src)
+	dug = 1
+	//icon_plating = "asteroid_dug"
+	icon_state = "asteroid_dug"
+	return
+
+/turf/simulated/floor/asteroid/proc/updateMineralOverlays()
+	src.overlays.len = 0
+	spawn(1)
+		for(var/dir in cardinal)
+			if(istype(get_step(src,dir), /turf/unsimulated/mineral))
+				var/turf/unsimulated/mineral/M = get_step(src,dir)
+				overlays += image('icons/turf/walls.dmi', M.return_icon_state(dir))
+
+/turf/simulated/floor/asteroid/proc/fullUpdateMineralOverlays()
+
```

Comment:
Literally copypasted from /turf/unsimulated/floor/asteroid.

Be angry about that